storage/enrollment.go

```diff
@@ -0,0 +1,92 @@
+package storage
+
+import (
+	"context"
+	"time"
+
+	"github.com/micromdm/nanomdm/mdm"
+)
+
+// EnrollmentsQueryFilter is a filter for enrollments. All query parameters are optional.
+// If multiple parameters are provided, they are combined with logical AND.
+// e.g. if both EnrollmentTypes and Serials are provided, only enrollments matching
+// both criteria are returned.
+type EnrollmentsQueryFilter struct {
+	IDs            []string `json:"ids,omitempty"`
+	Serials        []string `json:"serials,omitempty"`
+	UserShortNames []string `json:"user_short_names,omitempty"`
+	Types          []string `json:"types,omitempty"`
+	Enabled        *bool    `json:"enabled,omitempty"`
+}
+
+// Pagination supports both limit/offset, as well as cursor.
+// For offset-based queries, Limit and Offset must be set. Use Offset = 0 for the first request.
+// For cursor-based queries, Limit and Cursor must be set. Use Cursor = "" for the first request.
+type Pagination struct {
+	Limit  int
+	Offset *int
+	Cursor string
+}
+
+type EnrollmentQueryOptions struct {
+	// By default we do not include the Device Identity certificate in the response.
+	IncludeDeviceCert bool `json:"include_device_cert,omitempty"`
+}
+
+// EnrollmentsQuery represents a query for MDM enrollments.
+type EnrollmentsQuery struct {
+	Filter     EnrollmentsQueryFilter `json:"filter,omitempty"`
+	Pagination Pagination             `json:"-"`
```

Comment:
Possible for some backends to not even need pagination and have reasonable defaults.

```suggestion
	Pagination *Pagination             `json:"pagination,omitempty"`
```